Add `author_url` to embed command

aphkyle opened the pull request:
quick fix to /discord

aphkyle committed:
add `author_url`

gustavwilliam reviewed (changes requested):
Thanks for the PR!

I tried doing `/discord embed title: hello description: there author: hello author_url: 0`. This raises: 
```py
disnake.ext.commands.errors.CommandInvokeError: Command raised an exception: HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
In data.embeds.0.author.url: Scheme "0" is not supported. Scheme must be one of ('http', 'https').
```
Let's add some error handling for strange URLs.

aphkyle committed:
error handle

aphkyle commented:
> Thanks for the PR!
> 
> I tried doing `/discord embed title: hello description: there author: hello author_url: 0`. This raises:
> 
> ```python
> disnake.ext.commands.errors.CommandInvokeError: Command raised an exception: HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
> In data.embeds.0.author.url: Scheme "0" is not supported. Scheme must be one of ('http', 'https').
> ```
> 
> Let's add some error handling for strange URLs.

Added error handler :D

gustavwilliam commented on bot/exts/discord/discord.py:
```diff
@@ -4,7 +4,7 @@
 from bot.utils.embeds import create_embed
 from bot.utils.color import parse_color, rgb_to_hex
-from disnake import User
+from disnake import HTTPException, User
```
Why is this imported, since `disnake.HTTPException` is being used, instead of `HTTPException`?

aphkyle replied:
this was an accident, it is required to use `disnake.HTTPException` for the match case

gustavwilliam commented on bot/exts/discord/discord.py:
```diff
@@ -74,16 +76,30 @@ async def embed(
+            # instead of just `HTTPException`
+            case disnake.HTTPException:
+                raise commands.BadArgument(
+                    "Invalid URL responded."
```
```suggestion
                    "Invalid URL provided."
```
Maybe something like this? Or could we be more specific, like what attribute is creating this issue?

gustavwilliam commented on bot/exts/discord/discord.py:
```diff
@@ -74,16 +76,30 @@ async def embed(
+                )
+            case _:
+                raise commands.BadArgument(
+                    "No valid embed could be generated form the given input."
```
```suggestion
                    "No valid embed could be generated from the given input."
```

gustavwilliam commented on bot/exts/discord/discord.py:
```diff
@@ -74,16 +76,30 @@ async def embed(
+        inter: ApplicationCommandInteraction,
+        error: commands.CommandInvokeError,
+    ) -> None:
+        """Error event of command `embed`"""
```
```suggestion
        """Error event handler for `embed` command."""
```
What do you think?

aphkyle committed:
Update bot/exts/discord/discord.py

Co-authored-by: Gustav Odinger <65498475+gustavwilliam@users.noreply.github.com>

aphkyle committed:
Update bot/exts/discord/discord.py

Co-authored-by: Gustav Odinger <65498475+gustavwilliam@users.noreply.github.com>

aphkyle committed:
Update bot/exts/discord/discord.py

Co-authored-by: Gustav Odinger <65498475+gustavwilliam@users.noreply.github.com>

aphkyle committed:
Remove unneeded import

gustavwilliam reviewed (approved):
Code-wise, LGTM. I'll just do some quick testing when I get home, but I don't really expect to encounter any issues with this.

gustavwilliam committed:
Fix bug where match statement didn't match error

In order to match a class (like HTTPException), it needs to use (), as
in HTTPException(), instead of just HTTPException.

The command that's changed also needs a `self` argument.

gustavwilliam committed:
Fix typing of embed error handler

gustavwilliam commented:
@aphkyle there was a bug or two in the code, but I (hopefully) fixed them while reviewing.

Thanks a lot for the PR 😄
